common/nng/socket.cpp

```diff
@@ -39,14 +39,22 @@ Socket::~Socket()
 
 bool Socket::SyncRequired() { return false; }
 
-int Socket::Connect(const std::string &addr, bool retry)
+int Socket::Connect(const std::string &addr, int retry)
```

Comment:
Since we expect retry to be 0, 1 or a larger number, I would make it
```suggestion
int Socket::Connect(const std::string &addr, unsigned retry)
```